lib/charms/data_platform_libs/v0/data_models.py

```diff
@@ -231,10 +233,15 @@ def write(relation_data: RelationDataContent, model: BaseModel):
         relation_data: pointer to the relation databag
         model: instance of pydantic model to be written
     """
-    for key, value in model.dict(exclude_none=True).items():
-        relation_data[key.replace("_", "-")] = (
-            str(value) if isinstance(value, str) or isinstance(value, int) else json.dumps(value)
-        )
+    for key, value in model.dict(exclude_none=False).items():
```

Comment:
I am not sure what version of `pydantic` you are using (I could not find this in the `requirements.txt`, `tox.ini` etc?), however if using version `2.` you might want to use `model_dump` instead of `dict` as it has a deprecation warning.

Reply:
This is a library managed by the data platform team and not a part of my code. I'm not able to optimise it.